src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java

```diff
@@ -97,14 +97,89 @@ public Map<String, ClinicalDataType> getClinicalAttributeDatatypeMap() {
     
     @Override
     public List<ClinicalDataCountItem> getClinicalDataCounts(StudyViewFilter studyViewFilter, List<String> filteredAttributes) {
-        return studyViewRepository.getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes)
+        Map<String, ClinicalDataType> clinicalAttributeDatatypeMap = this.getClinicalAttributeDatatypeMap();
+        StudyViewFilterContext studyViewFilterContext = createContext(studyViewFilter);
+            
+        int filteredSamplesCount = studyViewRepository.getFilteredSamplesCount(studyViewFilterContext);
+        int filteredPatientsCount = studyViewRepository.getFilteredPatientCount(studyViewFilterContext);
+        
+        List<ClinicalDataCountItem> clinicalDataCountItems = studyViewRepository
+            .getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes)
             .stream().collect(Collectors.groupingBy(ClinicalDataCount::getAttributeId))
             .entrySet().parallelStream().map(e -> {
                 ClinicalDataCountItem item = new ClinicalDataCountItem();
                 item.setAttributeId(e.getKey());
                 item.setCounts(e.getValue());
                 return item;
-            }).collect(Collectors.toList());
+            }).toList();
+
+        // This rest of the code is postprocessing to adjust NA counts for multiple studies
```

Comment:
Can we break this up into pieces... this one function is just under 100 lines long